File: SnakeWear/SnakeModel/Direction.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SnakeModel
{
    public enum Directions { LEFT, UP, RIGHT, DOWN };

    public class Direction
    {
        public int X { get; }
        public int Y { get; }

        public Direction(Directions direction)
        {
            switch (direction)
            {
                case Directions.LEFT:
                    X = -1;
                    Y = 0;
                    break;
                case Directions.UP:
                    X = 0;
                    Y = -1;
                    break;
                case Directions.RIGHT:
                    X = 1;
                    Y = 0;
                    break;
                case Directions.DOWN:
                    X = 0;
                    Y = 1;
                    break;
            }
        }

        private Direction(int x, int y)
        {
            X = x;
            Y = y;
        }

        public Direction Clone()
        {
            return new Direction(X, Y);
        }
    }
}

File: java/src/model/Direction.java
package model;

public class Direction {
    public enum Directions {LEFT, UP, RIGHT, DOWN}

    private int x, y;

    public Direction(Directions direction) {
        switch (direction) {
            case LEFT:
                x = -1;
                y = 0;
                break;
            case UP:
                x = 0;
                y = -1;
                break;
            case RIGHT:
                x = 1;
                y = 0;
                break;
            case DOWN:
                x = 0;
                y = 1;
                break;
        }
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    private Direction(int x, int y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public Direction clone() {
        return new Direction(x, y);
    }
}